### ISSUE TITLE
Classification data out

### ISSUE THREAD

**Issue by mkosmala** [zooniverse/Panoptes#611]
I assume the GUI for science teams getting classification data out of Panoptes is a ways down the line. ( @brian-c right? ) What will be the best way to extract classification data from the back-end API? "List all classifications" from the Classification Collection?

**Comment by camallen** [comment]
@ggdhines thoughts?

**Comment by camallen** [comment]
@mkosmala - Raw classification data is probably best retrieved via the `/api/classifications?project_id=X` for your project. You'll need to have owner access to the project to get these classifications. This is not an optimal solution and we will be reviewing how this works in the future, most likely using a different database technology and/or perhaps creating downloadable archives of classification data links. 

I asked @ggdhines to contribute here because the aim is to have an aggregation engine(AE) provide you with a form of aggregation over the classification set. What kind of aggregation (consensus) algorithm are you thinking of running for your project? 

Finally any data that is produced by the AE will be available on the `/aggregations?workflow_id=X&subject_id=Y` (subject_id is a more restrictive filter here). FYI - I've still got to document this end point.

**Comment by ggdhines** [comment]
Both the aggregation results and the individual classifications should be available to the science teams. While I hope to have useful algorithms for each project, the science teams should obviously be able to access the raw classifications to do their own analysis.

**Comment by mkosmala** [comment]
I'm happy to have raw classifications, as some of what I'll be figuring out for this project is aggregation algorithms. We have several different types of questions and each will need its own aggregation method. The "draw an oval around a tree" question can likely use whatever standard aggregating algorithm you use for drawing tasks. But I also have set up a paired question where I'm actually going to use the distribution from the answers as part of the inference as to the "right" answer.

So, in short, raw classifications are fine (and preferred) for me.

**Comment by camallen** [comment]
@mkosmala if we create a request to construct an archive on demand, what format is best? JSON data (serialised api response) or a csv format?

It would work like this: the project owner requests a classification data archive dump to be created. Background workers would construct a json/csv? archive, upload it to s3 as an expirable asset and then mail the requesting user with a link.

**Comment by mkosmala** [comment]
@camallen That would be perfect! As to JSON vs. csv I'm agnostic. Both are easy to parse in Python.
